### PR TITLE
#161278870 Users should be able to view and edit their profiles

### PR DESCRIPTION
**Pull request by botnetdobbs**
#### What does this PR do?
- Adds functionality for users to view and update their profiles
#### Description of Task to be completed?
- Users can view their profiles
- Users can update their profiles
#### How should this be manually tested?
   `$ git clone https://github.com/andela/ah-sealteam-frontend.git`
   `$ git checkout feature/Users-should-be-able-to-view-and-edit-their-profiles-#161278870`
   `$ npm install`
   `$ npm start`
   `$ Register account via http://localhost:3000/signup'
- This will redirect you to the profile page `http://localhost:3000/profile`
#### What are the relevant pivotal tracker stories?
[#161278870](https://www.pivotaltracker.com/n/projects/2205273/stories/161278870)

### CODE REVIEW

**Comment by Adoniswalker** [comment]
I was able to access the page and I was not logged in. Also got this when i tried to update a profile. @botnetdobbs may be you should rebase on login and get private routes
screen shot 2018-12-13 at 09 46 21

**Comment by susanwere** [comment]
> I was able to access the page and I was not logged in. Also got this when i tried to update a profile. @botnetdobbs may be you should rebase on login and get private routes
> screen shot 2018-12-13 at 09 46 21

Plus One on this

**Comment by botnetdobbs** [comment]
> I was able to access the page and I was not logged in. Also got this when i tried to update a profile. @botnetdobbs may be you should rebase on login and get private routes
> screen shot 2018-12-13 at 09 46 21

The issue I'm having is that the navbar is not global. I.e I think it should have been incuded in the routes and also there is no function to identify a user as logged in. There are also conflicts with the CSS. I've used Bootstrap and some of it's classes seems to have been overriden. I'm working on solving...,

**Comment by botnetdobbs** [comment]
> > I was able to access the page and I was not logged in. Also got this when i tried to update a profile. @botnetdobbs may be you should rebase on login and get private routes
> > screen shot 2018-12-13 at 09 46 21
> 
> Plus One on this

The username cannot be found because the user is not logged in hence no access token to be decoded. The issue is with the private routes which is not available for now but @Adoniswalker has it done in his PR. So I'm waiting for it to be merged so I can rebase and update my feature...,

**Comment by botnetdobbs** [comment]
@Adoniswalker @Bluey95 I've implemented a private router. A user cannot access the profile without first logging in and hence the issues raised above should be fixed. Kindly review. Your feedback will be appreciated

File: .editorconfig [hunk 1]
```diff
@@ -11,24 +11,24 @@ charset = utf-8
...
 [*.js]
 indent_style = space
-indent_size = 2
+indent_size = 4
 
 [*.hbs]
 insert_final_newline = false
 indent_style = space
-indent_size = 2
+indent_size = 4
```
Comment (Adoniswalker) [inline .editorconfig]: Indent size should be 2 for this, html and css

File: src/RootReducer.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import { combineReducers } from 'redux';
 import allErrors from './reducers/SignUpReducer';
 import userAuth from './reducers/SigninReducer';
+import userInfo from './reducers/ProfileReducer';
 
 export default combineReducers({
   userAuth,
-  allErrors
+  allErrors,
```
Comment (Adoniswalker) [inline src/RootReducer.js]: When rebasing move this info to reducers/index.js

File: src/reducers/RootReducer.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { combineReducers } from 'redux';
```
Comment (Adoniswalker) [inline src/reducers/RootReducer.js:1]: You will delete this file and move this to reducer/index.js

**Comment by botnetdobbs** [comment]
@Adoniswalker Thanks for the review. I've rebased and fixed the above-mentioned issues.

**Review by Adoniswalker — APPROVED** [review]

**Review by gtsofa — APPROVED** [review]

**Review by asheux — APPROVED** [review]

**Review by rickynyairo — APPROVED** [review]

**Pull request merged**